knotx-core/knotx-launcher/src/test/java/com/cognifide/knotx/launcher/SystemPropsConfigurationTest.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Knot.x - Reactive microservice assembler - Launcher
+ *
+ * Copyright (C) 2016 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.cognifide.knotx.launcher;
+
+import static org.junit.Assert.assertEquals;
+
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.ClearSystemProperties;
+
+public class SystemPropsConfigurationTest {
+
+  private final static String EXPECTED_OBJECT = "{\"someInteger\":1234,\"someBoolean\":false,\"someString\":\"foo\","
+      + "\"someNested\":{\"nestedInteger\":1234,\"nestedBoolean\":false,\"nestedString\":\"foo\"}}";
+
+
+  @Rule
+  public final ClearSystemProperties cleanedProperties = new ClearSystemProperties("io.knotx.KnotxServer",
+      "io.knotx.KnotxServer.options.config.httpPort", "io.knotx.KnotxServer.options.config.pass",
+      "io.knotx.KnotxServer.options.config.settings.bogus", "io.knotx.KnotxServer.options.config", "io.knotx.KnotxServer.options.haGroup",
+      "io.knotx.KnotxServer.options.config.someData");
+
+  @Test
+  public void setNoEnvValue() {
```

Comment:
Small refactor of test method names required according to [Test Naming Convention](https://github.com/Cognifide/knotx/blob/master/CONTRIBUTING.md#tests-naming-convention)